src/parser/Expression.ts

```diff
@@ -124,6 +125,7 @@ export class FunctionExpression extends Expression implements TypedefProvider {
     readonly symbolTable: SymbolTable;
 
     constructor(
+        name: Identifier,
```

Comment:
I don't think `name` belongs here. If a function has a name, the `FunctionStatement` has that name. Otherwise, it's an anonymous function expression and doesn't actually have a name. So in either case, `FunctionExpression` shouldn't have a name.

If you must have the name, then we should pass in the `FunctionStatement` that owns this expression.